### ISSUE TITLE
[MNG-6172] Precedence of command-line system property options has changed

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7271]
**[Stuart McCulloch](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcculls)** opened **[MNG-6172](https://issues.apache.org/jira/browse/MNG-6172?redirect=false)** and commented

The current state of master (what will eventually become 3.5.0) has reversed the precedence of command-line system property options compared to previous releases of Maven.

For example, running this command with a basic project:

```
mvn -Dmaven.repo.local=/tmp/aaa -Dmaven.repo.local=/tmp/zzz validate
```

using current master will cause `/tmp/aaa` to be created (first-one-wins), whereas for all previous releases of Maven `/tmp/zzz` would have been created (last-one-wins)

This has the potential to break CI builds which relied on the previous last-one-wins behaviour.

This was introduced by the fix for MNG-6078  https://github.com/apache/maven/commit/ca4303031357a7decaee8de770b71fb2c2fedd28 - by reversing the whole array the precedence between options on the same command line has been reversed, not just the relationship between .mvn/maven.config options and command line options.

---

**Backported to:** [3.5.0-alpha-1](https://github.com/apache/maven/milestone/92?closed=1)

1 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Do we need to drop this commit or rework?

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

I think we need to fix this. This would catch me out

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

Fixed with dc9c4db4494b62e2231bb67b39678decf6329852 in maven-core
